src/net_processing.cpp

```diff
@@ -2582,16 +2588,25 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
         return;
     }
 
-    if (msg_type == NetMsgType::ADDR) {
+    if (msg_type == NetMsgType::ADDR || msg_type == NetMsgType::ADDRV2) {
+        int stream_version = vRecv.GetVersion();
+        if (msg_type == NetMsgType::ADDRV2) {
+            // Add ADDRV2_FORMAT to the version so that the CNetAddr and CAddress
+            // unserialize methods know that an address in v2 format is coming.
+            stream_version |= ADDRV2_FORMAT;
```

Comment:
`addr` and `addrv2` are just a transport encodings. Once received, the address is not in any particular encoding, it is a `CAddress` object in the memory of the node, not in `addr` nor in `addrv2` encoding.

So, when we further relay a `CAddress` object we don't care in what encoding we received it.

Reply:
> So, when we further relay a CAddress object we don't care in what encoding we received it.

Right, it's a just a matter of stating this in the BIP to avoid other implementations not propagating addresses due to receiving transport encoding A but forgetting to relay the decoded addr stream to peers speaking transport encoding B.

Reply:
I don't think we should aim to specify relay behavior in BIPs. There is no requirement that all software behaves the same, and best practices around it are constant changing in response to attacks and observed network conditions.

The BIP specifies what means of communicating addresses are available. What that gets used for is local node policy.